src/shared/components/UserList.js

```diff
@@ -8,27 +8,59 @@ import ProfilePicture from "~/shared/components/ProfilePicture";
 import ConfirmModal from "~/shared/components/ConfirmModal";
 import { fullName, currentUser } from "~/util";
 import { modalProps } from "~/util/modal";
-import styles from "~/shared/styles/userList";
+import styles, { nameFontSize } from "~/shared/styles/userList";
 
 const RadiumGlyphicon = Radium(Glyphicon);
 
+const userLabelHeight = 70;
+
 @Radium
 class UserLabel extends React.Component {
+
     state = { isModalOpen: false };
+
+    adjustFontSize = () => {
+        let fontSize = parseInt(this.refs[this.props.user._id]
+            .style.fontSize.slice(0, -2) // removes "px" at end
+        );
+        while (this.refs[this.props.user._id]
+            .parentElement.clientHeight <= userLabelHeight
+            && fontSize < nameFontSize
+        ) {
+            fontSize++;
+            this.refs[this.props.user._id]
+                .style.setProperty("font-size", fontSize + "px");
+        } // makes font as large as possible under the default size while keeping box from expanding
+        while (this.refs[this.props.user._id]
+            .parentElement.clientHeight > userLabelHeight
+        ) {
+            fontSize--;
+            this.refs[this.props.user._id]
+                .style.setProperty("font-size", fontSize + "px");
+        } //makes font small enough so the box is not expanded
+    }
+
+    componentDidMount = () => {
+        this.adjustFontSize();
+        window.addEventListener("resize", this.adjustFontSize);
+    }
+
     render() {
         return (
             <Col sm={6} md={4} lg={3}>
                 <span
                     style={styles.userDisplay.span}
-                    onClick={() => window.location.assign("/profiles/id/" + user._id)}
+                    onClick={() =>
+                        window.location.assign("/profiles/id/" + this.props.user._id)
+                    }
                 >
                     <ProfilePicture
                         user={this.props.user}
                         picSize="small"
                         frameSize={30}
                         hasIndicator
                     />
-                    <span style={styles.userDisplay.name}>
+                    <span style={styles.userDisplay.name} ref={this.props.user._id}>
```

Comment:
Refs are per-component, so there is no reason to use the _id as a ref here instead of a constant string.